src/game_launcher.cpp

```diff
@@ -820,23 +821,29 @@ bool game_launcher::goto_editor()
 
 void game_launcher::start_wesnothd()
 {
-	std::string wesnothd_program = preferences::get_mp_server_program_name().empty()
-		? filesystem::get_exe_dir() + "/" + filesystem::get_program_invocation("wesnothd")
-		: preferences::get_mp_server_program_name();
-
-	std::string config = filesystem::get_user_config_dir() + "/lan_server.cfg";
-	if (!filesystem::file_exists(config)) {
+	std::string path = filesystem::get_user_config_dir() + "/lan_server.cfg";
+	if (!filesystem::file_exists(path)) {
 		// copy file if it isn't created yet
-		filesystem::write_file(config, filesystem::read_file(filesystem::get_wml_location("lan_server.cfg")));
+		filesystem::write_file(path, filesystem::read_file(filesystem::get_wml_location("lan_server.cfg")));
+	}
+
+	config lan;
+	read(lan, path);
+
+	std::string wesnothd_program = lan["exe"].str("");
```

Comment:
I think it might be better to call this something like "binary" or "command".

Is making it be a shell script supported? If so, "command" is probably better; if not, I'd lean more towards "binary". It should also be made clear whether it can be a shell _command_ (as in, whether this string will itself be parsed as a command). From the code I _think_ it looks like it's not? I'm not quite sure, though.

Reply:
Renamed to binary - a shell command would fail the file exists check.

Reply:
A shell _command_ would, but not a shell _script_.

Reply:
Ah, then yeah, a shell script would probably work as well. Changed to command then.

Reply:
The distinction (that it must be a valid file path and is not interpreted by the shell) can just be put in documentation somewhere.

Reply:
Added to the help output of wesnothd and the wesnothd man page.